x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -795,16 +807,30 @@ public int populateAndRead(
             final RangeAvailableHandler reader,
             final RangeMissingHandler writer
         ) throws Exception {
+            // We are interested in the total time that the system spends when fetching a result (including time spent queuing), so we start
+            // our measurement here.
+            final long startTime = threadPool.relativeTimeInMillis();
```

Comment:
@original-brownbear We discussed moving the instrumentation into the calling code so as to account for time spent waiting on threadpool queues, etc. After our discussion, I realized that the same effect can be achieved by simply starting the timer outside the closure.  This has the added benefit of not introducing a leaky abstraction/api which would otherwise have placed the onus of setting-up the instrumentation on the caller.

WDYT?

Reply:
Hmm indeed :) This does look correct. You win at last, fine by me to do it this way!